service/src/main/java/bio/terra/workspace/service/workspace/model/GcpCloudContext.java

```diff
@@ -19,4 +21,36 @@ public String getGcpProjectId() {
   public ApiGcpContext toApi() {
     return new ApiGcpContext().projectId(getGcpProjectId());
   }
+
+  public static final long GCP_CLOUD_CONTEXT_DB_VERSION = 1;
+
+  public String serialize() {
+    GcpCloudContextV1 dbContext = GcpCloudContextV1.from(getGcpProjectId());
+    return DbSerDes.toJson(dbContext);
+  }
+
+  public static GcpCloudContext deserialize(String json) {
+    if (json == null) {
+      return null;
+    }
+
+    GcpCloudContextV1 result = DbSerDes.fromJson(json, GcpCloudContextV1.class);
+    if (result.version != GCP_CLOUD_CONTEXT_DB_VERSION) {
+      throw new InvalidSerializedVersionException("Invalid serialized version");
+    }
+    return new GcpCloudContext(result.gcpProjectId);
+  }
+
+  public static class GcpCloudContextV1 {
+    /** Version marker to store in the db so that we can update the format later if we need to. */
+    @JsonProperty public final long version = GCP_CLOUD_CONTEXT_DB_VERSION;
```

Comment:
if you do it this way, how will you define it for the V2 class?

Reply:
It will have to have a `long version`. Or are you asking about the naming of the static?
I just used what was there.